wagtail/contrib/wagtailapi/api.py

```diff
@@ -0,0 +1,535 @@
+from __future__ import absolute_import
+
+import json
+import urllib
+from functools import wraps
+from collections import OrderedDict
+
+from modelcluster.models import get_all_child_relations
+from taggit.managers import _TaggableManager
+from taggit.models import Tag
+
+from django.db import models
+from django.utils.encoding import force_text
+from django.http import HttpResponse, HttpResponseBadRequest, HttpResponseNotFound, Http404
+from django.shortcuts import get_object_or_404
+from django.core.paginator import Paginator, EmptyPage
+from django.core.serializers.json import DjangoJSONEncoder
+from django.conf.urls import url
+from django.conf import settings
+
+from wagtail.wagtailcore.models import Page
+from wagtail.wagtailimages.models import get_image_model
+from wagtail.wagtaildocs.models import Document
+from wagtail.wagtailcore.utils import resolve_model_string
+from wagtail.wagtailsearch.backends import get_search_backend
+
+from .utils import get_base_url
+
+
+class WagtailAPIJSONEncoder(DjangoJSONEncoder):
+    def default(self, o):
+        if isinstance(o, _TaggableManager):
+            return list(o.all())
+        elif isinstance(o, Tag):
+            return o.name
+        else:
+            return super(WagtailAPIJSONEncoder, self).default(o)
+
+
+def get_api_data(obj, fields):
+    # Find any child relations (pages only)
+    child_relations = {}
+    if isinstance(obj, Page):
+        child_relations = {
+            child_relation.field.rel.related_name: child_relation.model
+            for child_relation in get_all_child_relations(type(obj))
+        }
+
+    # Loop through fields
+    for field_name in fields:
+        # Check child relations
+        if field_name in child_relations and hasattr(child_relations[field_name], 'api_fields'):
+            yield field_name, [
+                dict(get_api_data(child_object, child_relations[field_name].api_fields))
+                for child_object in getattr(obj, field_name).all()
+            ]
+            continue
+
+        # Check django fields
+        try:
+            field = obj._meta.get_field_by_name(field_name)[0]
+            yield field_name, field._get_val_from_obj(obj)
+            continue
+        except models.fields.FieldDoesNotExist:
+            pass
+
+        # Check attributes
+        if hasattr(obj, field_name):
+            value = getattr(obj, field_name)
+            yield field_name, force_text(value, strings_only=True)
```

Comment:
Should this really use `force_text`? A model could have a boolean, numeric, or None value on an attribute that could be useful to send.